Build broken on macOS Big Sur 11.5.2

Cloning and building Julia on the current master (5bf1ca5c724143091973ef6632f1b3026d8021af)
```bash
git clone ...
make
```
leads to
```
...
    JULIA usr/lib/julia/sys-o.a
/bin/sh: line 1: 11329 Segmentation fault: 11  JULIA_BINDIR=/Users/jonas/Projects/julia2/usr/bin WINEPATH="/Users/jonas/Projects/julia2/usr/bin;$WINEPATH" /Users/jonas/Projects/julia2/usr/bin/julia -O3 -C "native" --output-o /Users/jonas/Projects/julia2/usr/lib/julia/sys-o.a.tmp --startup-file=no --warn-overwrite=yes --sysimage /Users/jonas/Projects/julia2/usr/lib/julia/sys.ji /Users/jonas/Projects/julia2/contrib/generate_precompile.jl 1
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/Users/jonas/Projects/julia2/usr/lib/julia/sys-o.a] Error 1
make: *** [julia-sysimg-release] Error 2
```
While working on my "old" clone, I encountered the same problem. `make clean` and `make cleanall` didn't fix the issue. I didn't try this on the new clone. I also tried creating a `Make.user` containing `JULIA_PRECOMPILE=0`, but that didn't work either.

I am running macOS 11.5.2 on a 2018 MacBook Pro. From my homebrew Julia installation:
```
julia> versioninfo()
Julia Version 1.6.2
Commit 1b93d53fc4 (2021-07-14 15:36 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```

On my local machine, with otherwise identical build environments, the build passes with Xcode 12 and fails with Xcode 13.

| macOS         | Xcode           | Clang                       | Build              |
| ------------- | --------------- | --------------------------- | -------------------|
| 11.6 (20G165) | 12.5.1 (12E507) | 12.0.5 (clang-1205.0.22.11) | :white_check_mark: |
| 11.6 (20G165) | 13.0 (13A233)   | 13.0.0 (clang-1300.0.29.3)  | :x                 |

```
Sysimage built. Summary:
Total ───────  73.074822 seconds
Base: ───────  29.993381 seconds 41.0448%
Stdlibs: ────  43.078845 seconds 58.9517%
    JULIA usr/lib/julia/sys-o.a
/bin/sh: line 1:  6432 Segmentation fault: 11  JULIA_BINDIR=/Users/dilum/Downloads/julia/usr/bin WINEPATH="/Users/dilum/Downloads/julia/usr/bin;$WINEPATH" /Users/dilum/Downloads/julia/usr/bin/julia -O3 -C "native" --output-o /Users/dilum/Downloads/julia/usr/lib/julia/sys-o.a.tmp --startup-file=no --warn-overwrite=yes --sysimage /Users/dilum/Downloads/julia/usr/lib/julia/sys.ji /Users/dilum/Downloads/julia/contrib/generate_precompile.jl 1
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/Users/dilum/Downloads/julia/usr/lib/julia/sys-o.a] Error 1
make: *** [julia-sysimg-release] Error 2
```

--- a/src/init.c
+++ b/src/init.c
@@ -628,6 +628,8 @@ static void restore_fp_env(void)
     }
 }
 
+static NOINLINE void _finish_julia_init(JL_IMAGE_SEARCH rel, jl_ptls_t ptls, jl_task_t *ct);
+
 JL_DLLEXPORT void julia_init(JL_IMAGE_SEARCH rel)
 {
     jl_init_timing();
@@ -722,9 +724,14 @@ JL_DLLEXPORT void julia_init(JL_IMAGE_SEARCH rel)
     jl_init_threading();
 
     jl_ptls_t ptls = jl_init_threadtls(0);
-    jl_init_root_task(ptls, stack_lo, stack_hi);
-    jl_task_t *ct = jl_current_task;
+    // warning: this changes `jl_current_task`, so be careful not to call that from this function
+    jl_task_t *ct = jl_init_root_task(ptls, stack_lo, stack_hi);
+    JL_GC_PROMISE_ROOTED(ct);
+    _finish_julia_init(rel, ptls, ct);
+}
 
+static NOINLINE void _finish_julia_init(JL_IMAGE_SEARCH rel, jl_ptls_t ptls, jl_task_t *ct)
+{
     jl_init_threadinginfra();
 
     jl_resolve_sysimg_location(rel);

--- a/src/julia_internal.h
+++ b/src/julia_internal.h
@@ -741,7 +741,7 @@ void jl_init_intrinsic_functions(void);
 void jl_init_intrinsic_properties(void);
 void jl_init_tasks(void) JL_GC_DISABLED;
 void jl_init_stack_limits(int ismaster, void **stack_hi, void **stack_lo);
-void jl_init_root_task(jl_ptls_t ptls, void *stack_lo, void *stack_hi);
+jl_task_t *jl_init_root_task(jl_ptls_t ptls, void *stack_lo, void *stack_hi);
 void jl_init_serializer(void);
 void jl_gc_init(void);
 void jl_init_uv(void);

--- a/src/partr.c
+++ b/src/partr.c
@@ -247,7 +247,9 @@ void jl_threadfun(void *arg)
     jl_ptls_t ptls = jl_init_threadtls(targ->tid);
     void *stack_lo, *stack_hi;
     jl_init_stack_limits(0, &stack_lo, &stack_hi);
-    jl_init_root_task(ptls, stack_lo, stack_hi);
+    // warning: this changes `jl_current_task`, so be careful not to call that from this function
+    jl_task_t *ct = jl_init_root_task(ptls, stack_lo, stack_hi);
+    JL_GC_PROMISE_ROOTED(ct);
     jl_install_thread_signal_handler(ptls);
 
     // set up sleep mechanism for this thread
@@ -262,7 +264,7 @@ void jl_threadfun(void *arg)
     free(targ);
 
     (void)jl_gc_unsafe_enter(ptls);
-    jl_finish_task(jl_current_task); // noreturn
+    jl_finish_task(ct); // noreturn
 }
 
 

--- a/src/task.c
+++ b/src/task.c
@@ -1251,7 +1251,7 @@ static char *jl_alloc_fiber(jl_ucontext_t *t, size_t *ssize, jl_task_t *owner) J
 #endif
 
 // Initialize a root task using the given stack.
-void jl_init_root_task(jl_ptls_t ptls, void *stack_lo, void *stack_hi)
+jl_task_t *jl_init_root_task(jl_ptls_t ptls, void *stack_lo, void *stack_hi)
 {
     assert(ptls->root_task == NULL);
     // We need `gcstack` in `Task` to allocate Julia objects; *including* the `Task` type.
@@ -1327,13 +1327,14 @@ void jl_init_root_task(jl_ptls_t ptls, void *stack_lo, void *stack_hi)
 #endif
         if (jl_setjmp(ptls->copy_stack_ctx.uc_mcontext, 0))
             start_task(); // sanitizer_finish_switch_fiber is part of start_task
-        return;
+        return ct;
     }
     ssize = JL_STACK_SIZE;
     char *stkbuf = jl_alloc_fiber(&ptls->base_ctx, &ssize, NULL);
     ptls->stackbase = stkbuf + ssize;
     ptls->stacksize = ssize;
 #endif
+    return ct;
 }
 
 JL_DLLEXPORT int jl_is_task_started(jl_task_t *t) JL_NOTSAFEPOINT

